DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountEditDetailsFormData.cs

```diff
@@ -33,6 +35,17 @@ public class MyAccountEditDetailsFormData : EditDetailsFormData, IValidatableObj
             Answer4 = delegateUser?.Answer4;
             Answer5 = delegateUser?.Answer5;
             Answer6 = delegateUser?.Answer6;
+
+            if (IsDelegateUser)
+            {
+                ProfessionalRegistrationNumber = delegateUser?.ProfessionalRegistrationNumber!;
+                if (delegateUser!.HasBeenPromptedForPrn)
+                {
+                    HasProfessionalRegistrationNumber = !string.IsNullOrEmpty(ProfessionalRegistrationNumber)
+                        ? YesNoSelectionEnum.Yes
+                        : YesNoSelectionEnum.No;
+                }
+            }

```

Comment:
It feels like this belongs in the Controller / Service really, because it's logic rather than just housing data.

Reply:
I've simplified this and moved the `logic` bits out to a helper. Could argue that it could go in a service.

Reply:
Ah, sorry, I completely missed that this is effectively in the viewModel constructor, in which case I think it's fine here... but going in the helper works too.